HIP/hip-631.md

```diff
@@ -0,0 +1,555 @@
+
+```protobuf
+message ContractCreateTransactionBody {
+	...
+    /**
+    * The 20-byte EVM address (differs from account default) to use for the given transaction. This address must be owned by the account.
+    */
+	bytes virtual_address_override = 21;
+}
+
+message ContractCallTransactionBody {
+	...
+    /**
+    * The 20-byte EVM address (differs from account default) to use for the given transaction. This address must be owned by the account.
+    */
+	bytes virtual_address_override = 5;
+}
+```
+
+Update the `TranscationRecord` to expose a new virtual address update when not present in the transaction 
+
+```protobuf
+message TransactionRecord {
+	...
+
+    /**
+    * The new default EVM address of the account created by this transaction.
+    * This field is populated only when the EVM address is not specified in the related transaction body.
+    */
+    bytes evm_address = 21;
+}
+```
+
+The following is a table of HAPI proto inputs and outputs to show case new/modified transaction calls and how details are exposed in record files
+
+| Transaction Type | Inputs (all keys and address map to valid ECDSA keys) | Account Exists | Record Output  (VirtualAddressUpdate) | Description |
+| --- | --- | --- | --- | --- |
+| ContractCreate | virtual_address_override | - | - | Contract create transaction with an override for the virtual address to be used |
+|  |  |  |  |  |
+| ContractCall | virtual_address_override | - | - | Contract call transaction with an override for the virtual address to be used |
+|  |  |  |  |  |
+| CryptoCreate | alias | N | evm_address = new | auto-create non-existing account with new virtual address. The provided virtual address will serve as the default virtual address for the account. |
+|  | key | N | evm_address = new | normal create non-existing account with new virtual address. The provided virtual address will serve as the default virtual address for the account. |
+|  | evm_address | N |  | lazy-create non-existing account with new virtual address. The provided virtual address will serve as the default virtual address for the account. |
+|  |  |  |  |  |
+| CryptoTransfer | accountId.alias | N | evm_address = new | transfer to non-existing account (auto-create with new virtual address). The provided virtual address will serve as the default virtual address for the account. |
+|  | accountId.alias | Y | - | transfer to existing |
+|  | accountId.evmAddress | N | evm_address = new | transfer to non-existing account (lazy-create with new virtual address). The provided virtual address will serve as the default virtual address for the account. |
+|  | accountId.evmAddress | Y | - | transfer to existing |
+|  |  |  |  |  |
+| CryptoUpdate | virtual_address_update.add.evmAddress = new (optional) virtualAddress.isDefault = true | Y | - | Add virtual address to existing account. Pull details from transaction body (not populating record preserves space) |
```

Comment:
We should have a row for disable as well.

Reply:
Added